runtime/doc/api.txt

```diff
@@ -1949,12 +1949,35 @@ nvim_get_option_info({name})                          *nvim_get_option_info()*
     • commalist: List of comma separated values
     • flaglist: List of single char flags
 
+    The last set information applies to the local value in the current buffer
+    or window if it is available, otherwise the global value information is
+    returned.
+
     Parameters: ~
       • {name}  Option name
 
     Return: ~
         Option Information
 
+nvim_get_option_info_ex({name}, {*opts})           *nvim_get_option_info_ex()*
+    Gets the option information for one option from arbitrary buffer or window
+
+    This function is similar to nvim_get_option_info except that it accepts an
+    additional options table {opts}, allowing to specify option scope, buffer
+    number or window used for retrieving the last set location information.
+
+    Parameters: ~
+      • {name}  Option name
+      • {opts}  Optional parameters
+                • scope: One of "global" or "local". Analogous to |:setglobal|
+                  and |:setlocal|, respectively.
+                • win: |window-ID|. Used for getting window local options.
+                • buf: Buffer number. Used for getting buffer local options.
+                  Implies {scope} is "local".
+
+    Return: ~
+        Option Information
```

Comment:
This is a full replacement for `nvim_get_option_info`, right? So let's move the docs for `nvim_get_option_info` here, and deprecate `nvim_get_option_info`.

Reply:
Will do that.